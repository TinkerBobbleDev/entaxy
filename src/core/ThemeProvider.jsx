import React from 'react'
import PropTypes from 'prop-types'
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import blueGrey from '@material-ui/core/colors/blueGrey'
import '../styles/globals.css'

const theme = createMuiTheme({
  palette: {
    primary: { main: blueGrey[800] },
    secondary: { main: '#1B9CC4' },
    info: {
      text: '#00acc1',
      background: '#00acc11a'
    },
    danger: {
      text: '#721c24',
      background: '#f8d7da66'
    }

  },
  typography: {
    useNextVariants: true
  }
})

const ThemeProvider = ({ children }) => (
  <CssBaseline>
    <MuiThemeProvider theme={theme}>
      {children}
    </MuiThemeProvider>
  </CssBaseline>
)

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
}

export default ThemeProvider
