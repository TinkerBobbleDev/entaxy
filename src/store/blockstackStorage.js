/* eslint-disable no-console */
import { getFile, putFile, isUserSignedIn } from 'blockstack'

const filename = 'entaxy.json'

export const loadState = () => {
  if (isUserSignedIn()) {
    return getFile(filename)
      .then(data => JSON.parse(data))
      .catch((error) => {
        console.log(`Error loading file ${filename}:`, error)
        return {}
      })
  }
  return Promise.resolve({})
}

export const saveState = (state) => {
  if (isUserSignedIn()) {
    return putFile(filename, JSON.stringify(state))
  }
  return undefined
}
