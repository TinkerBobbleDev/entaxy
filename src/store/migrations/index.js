import migration0 from './0'
import migration1 from './1'
import migration2 from './2'
import migration3 from './3'

export default {
  0: migration0,
  1: migration1,
  2: migration2,
  3: migration3
}
