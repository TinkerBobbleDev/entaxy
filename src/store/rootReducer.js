import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import settingsReducer from './settings/reducer'
import transactionsReducer from './transactions/reducer'
import marketValuesReducer from './marketValues/reducer'

export default combineReducers({
  user: userReducer,
  settings: settingsReducer,
  transactions: transactionsReducer,
  marketValues: marketValuesReducer
})
