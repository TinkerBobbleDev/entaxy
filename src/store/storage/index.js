/* eslint-disable import/no-cycle */
import { snakeCase } from 'lodash'
import * as blockstackStorage from './blockstackStorage'
import * as localStorage from './localStorage'
import { showOverlay, hideOverlay } from '../settings/actions'
import { initialState as settingsInitialState } from '../settings/reducer'
import { initialState as accountsInitialState } from '../accounts/reducer'
import { initialState as transactionsInitialState } from '../transactions/reducer'
import { initialState as exchangeRatesInitialState } from '../exchangeRates/reducer'
import { initialState as budgetInitialState } from '../budget/reducer'

export const loadItem = (itemName, newState) => (dispatch, getState) => {
  let payload = newState
  if (itemName === 'settings') {
    const { settings } = getState()
    if (settings !== null && settings !== undefined) {
      payload = { ...newState, overlayMessage: settings.overlayMessage }
    }
  }
  dispatch({
    type: `LOAD_${snakeCase(itemName).toUpperCase()}`,
    payload
  })
}

export const loadAllItems = (newState) => (dispatch) => {
  [
    'settings',
    'accounts',
    'transactions',
    'exchangeRates',
    'budget'
  ].map((itemName) => dispatch(loadItem(itemName, newState[itemName] || {})))
}


export const loadState = () => async (dispatch, getState) => {
  if (getState().user === undefined) return {}
  let newState
  switch (getState().user.isAuthenticatedWith) {
    case 'guest':
      dispatch(showOverlay('Loading data from localStorage...'))
      newState = await localStorage.loadState()
      break
    case 'blockstack':
      dispatch(showOverlay('Loading data from Blockstack...'))
      newState = await blockstackStorage.loadState()
      break
    // no default
  }
  if (newState === undefined) newState = { settings: settingsInitialState }
  dispatch(loadAllItems({
    ...newState,
    settings: {
      ...newState.settings,
      snackbarMessage: null
    }
  }))
  return dispatch(hideOverlay())
}

export const resetState = () => (dispatch, getState) => {
  const { settings } = getState()

  // Preserve the existing currency and locale
  dispatch(loadAllItems({
    settings: {
      ...settingsInitialState,
      currency: settings.currency,
      locale: settings.locale
    },
    accounts: accountsInitialState,
    transactions: transactionsInitialState,
    exchangeRates: exchangeRatesInitialState,
    budget: budgetInitialState
  }))
}

export const saveState = () => (_, getState) => {
  let { user } = getState()
  if (user === undefined) user = {}
  switch (user.isAuthenticatedWith) {
    case 'guest':
      return localStorage.saveState(getState())
    case 'blockstack':
      return blockstackStorage.saveState(getState())
    default:
      return undefined
  }
}
