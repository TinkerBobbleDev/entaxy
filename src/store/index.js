import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import rootReducer from './rootReducer'

const middlewares = [thunkMiddleware]

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger')
  middlewares.push(logger)
}

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
)

if (module.hot) {
  module.hot.accept(rootReducer, () => {
    store.replaceReducer(rootReducer)
  })
}

export default store
