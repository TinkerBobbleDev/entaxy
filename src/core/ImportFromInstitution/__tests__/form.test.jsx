import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '../../../store'
import ThemeProvider from '../../ThemeProvider'
import ImportFromInstitutionForm from '../form'

// jest.mock('../formFields', () => 'InstitutionFormFields')
describe.skip('Account form', () => {
  const mochHandleSubmit = jest.fn()
  const mochHandleChange = jest.fn()
  const mochHandleCancel = jest.fn()
  const mochHandleDelete = jest.fn()
  const accountGroup = {
    id: 'g1',
    apiKey: 'abc',
    apiSecret: 'def',
    accountIds: []
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot for regular account', () => {
    const component = renderer.create((
      <Provider store={store}>
        <ImportFromInstitutionForm
          handleSubmit={mochHandleSubmit}
          values={{}}
          errors={{}}
          touched={{}}
          isSubmitting={false}
          handleChange={mochHandleChange}
          handleCancel={mochHandleCancel}
          handleDelete={mochHandleDelete}
          classes={{ }}
          institution="TD"
        />
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for new Coinbase account', () => {
    const component = renderer.create((
      <Provider store={store}>
        <ThemeProvider>
          <ImportFromInstitutionForm
            handleSubmit={mochHandleSubmit}
            values={{}}
            errors={{}}
            touched={{}}
            isSubmitting={false}
            handleChange={mochHandleChange}
            handleCancel={mochHandleCancel}
            handleDelete={mochHandleDelete}
            classes={{ }}
            institution="Coinbase"
          />
        </ThemeProvider>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for existing Coinbase account', () => {
    const component = renderer.create((
      <Provider store={store}>
        <ThemeProvider>
          <ImportFromInstitutionForm
            handleSubmit={mochHandleSubmit}
            values={{}}
            errors={{}}
            touched={{}}
            isSubmitting={false}
            handleChange={mochHandleChange}
            handleCancel={mochHandleCancel}
            handleDelete={mochHandleDelete}
            classes={{ }}
            institution="Coinbase"
            accountGroup={accountGroup}
          />
        </ThemeProvider>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
