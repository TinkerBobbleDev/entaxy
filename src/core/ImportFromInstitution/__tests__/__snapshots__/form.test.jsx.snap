// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account form matches snapshot for existing Coinbase account 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 WithFormik(ImportFromInstitutionFormComponent)-root-228 MuiPaper-rounded"
  >
    <div
      className="WithFormik(ImportFromInstitutionFormComponent)-formHeader-229"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Import from
        Coinbase
      </h6>
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root WithFormik(ImportFromInstitutionFormComponent)-closeButton-230"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="MuiGrid-root WithFormik(ImportFromInstitutionFormComponent)-input-232 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <p
        className="MuiTypography-root MuiTypography-caption MuiTypography-paragraph"
      >
        Your browser will connect directly to 
        <strong>
          Coinbase
        </strong>
         so you're really importing your own data.
      </p>
      <div>
        <p
          className="MuiTypography-root MuiTypography-caption MuiTypography-paragraph"
        >
          <a
            href="https://www.coinbase.com/settings/api"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary"
            >
              Create an API key on Coinbase (opens in a new tab)
            </span>
          </a>
        </p>
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiCard-root  DescriptionCard-card-425 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root DescriptionCard-icon-426"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            <span
              className="MuiTypography-root MuiTypography-caption"
            >
              Make sure your api key has access to 
              <strong>
                all accounts
              </strong>
               and has the following permimisions:
              <ul>
                <li>
                  <strong>
                    wallet:accounts:read
                  </strong>
                </li>
                <li>
                  <strong>
                    wallet:transactions:read
                  </strong>
                </li>
              </ul>
            </span>
          </div>
        </div>
      </div>
    </div>
    <form
      className="WithFormik(ImportFromInstitutionFormComponent)-form-231"
      onSubmit={[Function]}
    >
      <span
        className="WithFormik(ImportFromInstitutionFormComponent)-form-231"
      >
        <div
          className="MuiFormControl-root MuiTextField-root WithFormik(ImportFromInstitutionFormComponent)-input-232"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            API key
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="API key"
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              maxLength={100}
              name="apiKey"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root WithFormik(ImportFromInstitutionFormComponent)-input-232"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            API secret
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="API secret"
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              maxLength={100}
              name="apiSecret"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </span>
      <hr
        className="MuiDivider-root"
      />
      <div
        className="WithFormik(ImportFromInstitutionFormComponent)-formActions-235"
      >
        <div
          className="SubmitButtonWithProgress-wrapper-488"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Import
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`Account form matches snapshot for new Coinbase account 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 WithFormik(ImportFromInstitutionFormComponent)-root-228 MuiPaper-rounded"
  >
    <div
      className="WithFormik(ImportFromInstitutionFormComponent)-formHeader-229"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Import from
        Coinbase
      </h6>
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root WithFormik(ImportFromInstitutionFormComponent)-closeButton-230"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="MuiGrid-root WithFormik(ImportFromInstitutionFormComponent)-input-232 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <p
        className="MuiTypography-root MuiTypography-caption MuiTypography-paragraph"
      >
        Your browser will connect directly to 
        <strong>
          Coinbase
        </strong>
         so you're really importing your own data.
      </p>
      <div>
        <p
          className="MuiTypography-root MuiTypography-caption MuiTypography-paragraph"
        >
          <a
            href="https://www.coinbase.com/settings/api"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary"
            >
              Create an API key on Coinbase (opens in a new tab)
            </span>
          </a>
        </p>
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiCard-root  DescriptionCard-card-425 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root DescriptionCard-icon-426"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            <span
              className="MuiTypography-root MuiTypography-caption"
            >
              Make sure your api key has access to 
              <strong>
                all accounts
              </strong>
               and has the following permimisions:
              <ul>
                <li>
                  <strong>
                    wallet:accounts:read
                  </strong>
                </li>
                <li>
                  <strong>
                    wallet:transactions:read
                  </strong>
                </li>
              </ul>
            </span>
          </div>
        </div>
      </div>
    </div>
    <form
      className="WithFormik(ImportFromInstitutionFormComponent)-form-231"
      onSubmit={[Function]}
    >
      <span
        className="WithFormik(ImportFromInstitutionFormComponent)-form-231"
      >
        <div
          className="MuiFormControl-root MuiTextField-root WithFormik(ImportFromInstitutionFormComponent)-input-232"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            API key
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="API key"
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              maxLength={100}
              name="apiKey"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root WithFormik(ImportFromInstitutionFormComponent)-input-232"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            API secret
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="API secret"
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              maxLength={100}
              name="apiSecret"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </span>
      <hr
        className="MuiDivider-root"
      />
      <div
        className="WithFormik(ImportFromInstitutionFormComponent)-formActions-235"
      >
        <div
          className="SubmitButtonWithProgress-wrapper-488"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Import
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`Account form matches snapshot for regular account 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 WithFormik(ImportFromInstitutionFormComponent)-root-1 MuiPaper-rounded"
  >
    <div
      className="WithFormik(ImportFromInstitutionFormComponent)-formHeader-2"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Import from
        TD
      </h6>
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root WithFormik(ImportFromInstitutionFormComponent)-closeButton-3"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="MuiGrid-root WithFormik(ImportFromInstitutionFormComponent)-input-5 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <p
        className="MuiTypography-root MuiTypography-caption MuiTypography-paragraph"
      >
        Your browser will connect directly to 
        <strong>
          TD
        </strong>
         so you're really importing your own data.
      </p>
    </div>
    <form
      className="WithFormik(ImportFromInstitutionFormComponent)-form-4"
      onSubmit={[Function]}
    >
      <hr
        className="MuiDivider-root"
      />
      <div
        className="WithFormik(ImportFromInstitutionFormComponent)-formActions-8"
      >
        <div
          className="SubmitButtonWithProgress-wrapper-198"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Import
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
